mongodbatlas/fw_provider.go

```diff
@@ -0,0 +1,394 @@
+	PrivateKey           types.String `tfsdk:"private_key"`
+	BaseURL              types.String `tfsdk:"base_url"`
+	RealmBaseURL         types.String `tfsdk:"realm_base_url"`
+	IsMongodbGovCloud    types.Bool   `tfsdk:"is_mongodbgov_cloud"`
+	AssumeRole           types.List   `tfsdk:"assume_role"`
+	SecretName           types.String `tfsdk:"secret_name"`
+	Region               types.String `tfsdk:"region"`
+	StsEndpoint          types.String `tfsdk:"sts_endpoint"`
+	AwsAccessKeyID       types.String `tfsdk:"aws_access_key_id"`
+	AwsSecretAccessKeyID types.String `tfsdk:"aws_secret_access_key"`
+	AwsSessionToken      types.String `tfsdk:"aws_session_token"`
+}
+
+type tfAssumeRoleModel struct { //nolint:govet //prefer to maintain order of attributes over small byte optimization
+	Duration          types.String `tfsdk:"duration"`
+	DurationSeconds   types.Int64  `tfsdk:"duration_seconds"`
+	ExternalID        types.String `tfsdk:"external_id"`
+	Policy            types.String `tfsdk:"policy"`
+	PolicyARNs        types.Set    `tfsdk:"policy_arns"`
+	RoleARN           types.String `tfsdk:"role_arn"`
+	SessionName       types.String `tfsdk:"session_name"`
+	SourceIdentity    types.String `tfsdk:"source_identity"`
+	Tags              types.Map    `tfsdk:"tags"`
+	TransitiveTagKeys types.Set    `tfsdk:"transitive_tag_keys"`
+}
+
+func (p *MongodbtlasProvider) Metadata(ctx context.Context, req provider.MetadataRequest, resp *provider.MetadataResponse) {
+	resp.TypeName = "mongodbatlas"
+	// resp.Version = "" we can look into defining this value automatically for release process
+}
+
+func (p *MongodbtlasProvider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Blocks: map[string]schema.Block{
+			"assume_role": fwAssumeRoleSchema,
+		},
+		Attributes: map[string]schema.Attribute{
+			"public_key": schema.StringAttribute{
+				Optional:    true,
+				Description: "MongoDB Atlas Programmatic Public Key",
+			},
+			"private_key": schema.StringAttribute{
+				Optional:    true,
+				Description: "MongoDB Atlas Programmatic Private Key",
+				Sensitive:   true,
+			},
+			"base_url": schema.StringAttribute{
+				Optional:    true,
+				Description: "MongoDB Atlas Base URL",
+			},
+			"realm_base_url": schema.StringAttribute{
+				Optional:    true,
+				Description: "MongoDB Realm Base URL",
+			},
+			"is_mongodbgov_cloud": schema.BoolAttribute{
+				Optional:    true,
+				Description: "MongoDB Atlas Base URL default to gov",
+			},
+			"secret_name": schema.StringAttribute{
+				Optional: true,
+			},
+			"region": schema.StringAttribute{
+				Optional: true,
+			},
+			"sts_endpoint": schema.StringAttribute{
+				Optional: true,
+			},
+			"aws_access_key_id": schema.StringAttribute{
+				Optional: true,
+			},
+			"aws_secret_access_key": schema.StringAttribute{
+				Optional: true,
+			},
+			"aws_session_token": schema.StringAttribute{
+				Optional: true,
+			},
```

Comment:
let's add a description for all the fields

Reply:
Added descriptions based off of the following documentation: https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs#aws-secrets-manager